ttygif.c

```diff
@@ -233,7 +239,12 @@ ttyplay (FILE *fp, ReadFunc read_func, WriteFunc write_func, Options o)
         }
 
         if (!skip && index != 0) {
-            if (sprintf(arg_buffer, " -delay %f %s", delay * 0.1, img_path) < 0) {
+            if (sprintf(arg_buffer, " -delay %f", delay * 0.1) < 0) {
+                fatalf("Error: Failed to format 'convert' parameters");
+            }
+            StringBuilder_write(sb, arg_buffer);
+            
+            if (sprintf(arg_buffer, " %s", img_path) < 0) {
```

Comment:
You can use a single call to `sprintf` to put the `delay` & `img_path` into the `arg_buffer`.